### PR TITLE
chore: use playwright test runner for `adapter-static` tests

### PR DESCRIPTION
**Pull request by benmccann**
This is a less code and easier to understand as it doesn't mix two test frameworks together. If we wanted to switch `uvu` with `vitest` this would be one less package to convert and one of the hardest ones to address

### CODE REVIEW

File: packages/adapter-static/test/apps/prerendered/test/test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as fs from 'node:fs';
+import { expect, test } from '@playwright/test';
+
+const cwd = process.cwd();
+
+test('generates HTML files', () => {
+	expect(fs.existsSync(`${cwd}/build/index.html`)).toBeTruthy;
```
Comment (gtm-nayan) [inline packages/adapter-static/test/apps/prerendered/test/test.js:7]:
```suggestion
	expect(fs.existsSync(`${cwd}/build/index.html`)).toBeTruthy();
```

File: packages/adapter-static/test/utils.js [hunk 2]
```diff
@@ -1,122 +1,46 @@
...
-	webkit: webkit
+const known_devices = {
+	chromium: devices['Desktop Chrome'],
+	firefox: devices['Desktop Firefox'],
+	webkit: devices['Desktop Safari']
 };
-const test_browser_name = /** @type {keyof typeof } */ (process.env.KIT_E2E_BROWSER ?? 'chromium');
+const test_browser = /** @type {keyof typeof known_devices} */ (
+	process.env.KIT_E2E_BROWSER ?? 'chromium'
+);
```
Comment (gtm-nayan) [inline packages/adapter-static/test/utils.js:10]: Looks like these tests aren't run cross-platform. For adapters I think the only cross-platform thing that needs to be tested would be dealing with Windows' path oddities.

Should we remove this part altogether or make the tests run cross-platform?

**Review by gtm-nayan — CHANGES_REQUESTED** [review]

File: packages/adapter-static/test/utils.js (hunk 2 above)
Comment (benmccann) [reply packages/adapter-static/test/utils.js]: ok. I went ahead and removed it for now

File: packages/adapter-static/test/apps/prerendered/test/test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as fs from 'node:fs';
+import { expect, test } from '@playwright/test';
+
+const cwd = process.cwd();
+
+test('generates HTML files', () => {
+	expect(fs.existsSync(`${cwd}/build/index.html`)).toBeTruthy();
+});
```
Comment (Rich-Harris) [inline packages/adapter-static/test/apps/prerendered/test/test.js:8]: feels kinda weird to use playwright for tests like these, no? should we split the tests up, or do you think that adds too much complexity?

Comment (benmccann) [reply packages/adapter-static/test/apps/prerendered/test/test.js:8]: yeah, it did feel slightly weird, but there are so few of them. I think it's simpler with just a single test runner. we can always revisit if we add more non-browser tests

Comment (Rich-Harris) [reply packages/adapter-static/test/apps/prerendered/test/test.js:8]: fair enough — that weirdness aside, this is much nicer

**Pull request merged**